lib/Runtime/Base/FunctionBody.h

```diff
@@ -316,13 +316,28 @@ namespace Js
     public:
         // These are public because we don't manage them nor their consistency;
         // the user of this class does.
-        void * address;
+        union address_type_ {
+            address_type_(void *ptr):asPtr(ptr) {}
+            address_type_(Js::JavascriptMethod method):asJsMethod(method) {}
 
-        ProxyEntryPointInfo(void* address, ThreadContext* context = nullptr):
+            void *asPtr;
+            Js::JavascriptMethod asJsMethod;
+        } address;
+        static_assert(sizeof(Js::JavascriptMethod) == sizeof(uintptr_t),
```

Comment:
IMO the issue was most likely an alignment related optimization issue. It don't see this static_assert is ever happening though.